components/kubelet/pod/component.go

```diff
@@ -108,35 +115,27 @@ func (c *component) CheckOnce() {
 		c.lastMu.Unlock()
 	}()
 
-	// assume "kubelet" is not installed, thus not needed to check its activeness
 	if c.checkDependencyInstalled == nil || !c.checkDependencyInstalled() {
+		// "kubelet" is not installed, thus not needed to check its activeness
 		return
 	}
 
-	// below are the checks in case "kubelet" is installed, thus requires activeness checks
-	cctx, ccancel := context.WithTimeout(c.ctx, 15*time.Second)
-	running := checkKubeletReadOnlyPortListening(cctx, c.kubeletReadOnlyPort)
-	ccancel()
-	if !running {
-		d.err = errors.New("kubelet is installed but kubelet is not running")
+	if c.checkKubeletRunning == nil || !c.checkKubeletRunning() {
+		// "kubelet" is not running, thus not needed to check its activeness
 		return
 	}
 
-	if c.checkServiceActive != nil {
-		var err error
-		cctx, ccancel = context.WithTimeout(c.ctx, 15*time.Second)
-		d.KubeletServiceActive, err = c.checkServiceActive(cctx)
-		ccancel()
-		if !d.KubeletServiceActive || err != nil {
-			d.err = fmt.Errorf("kubelet is installed but kubelet service is not active or failed to check (error %v)", err)
-			return
-		}
-	}
-
-	cctx, ccancel = context.WithTimeout(c.ctx, 30*time.Second)
+	// below are the checks in case "kubelet" is installed and running, thus requires activeness checks
+	cctx, ccancel := context.WithTimeout(c.ctx, 30*time.Second)
 	d.NodeName, d.Pods, d.err = listPodsFromKubeletReadOnlyPort(cctx, c.kubeletReadOnlyPort)
 	ccancel()
 
+	if d.err != nil {
+		c.failedCount++
```

Comment:
it's better to update the count in CheckOnce function.

Reply:
@eahydra This is already in CheckOnce function? https://github.com/leptonai/gpud/pull/589/files#diff-2aff32cd20735f318eff3f7d05a30e909e35b7e8176380e14127fb4827b8b5c4R107